loading.bs

```diff
@@ -0,0 +1,949 @@
+        text: Date header field; url: section-7.1.1.2
+spec: RFC8446; urlPrefix: https://tools.ietf.org/html/draft-ietf-tls-tls13-28#
+    text: ecdsa_secp256r1_sha256; type: dfn; url: section-4.2.3
+spec: draft-thomson-http-mice; urlPrefix: https://tools.ietf.org/html/draft-thomson-http-mice-02#
+    type: dfn
+        text: mi-sha256 parameter; url: section-3.1
+        text: integrity proof for the first record; url: section-2.2
+spec: draft-yasskin-wpack-bundled-exchanges; urlPrefix: https://wicg.github.io/webpackage/draft-yasskin-wpack-bundled-exchanges.html#
+    type: dfn
+        text: parsing a CBOR item; url: parse-known-length
+spec: draft-yasskin-http-origin-signed-responses; urlPrefix: https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#
+    type: dfn
+        text: stateful request header; url: stateful-headers
+        text: stateful response header; url: stateful-headers
+spec: draft-yasskin-httpbis-origin-signed-exchanges-impl-01; urlPrefix: https://tools.ietf.org/html/draft-yasskin-httpbis-origin-signed-exchanges-impl-01#
+    type: dfn
+        text: canonically-encoded CBOR; url: section-3.4
+        text: CanSignHttpExchanges; url: section-4.2
+        text: cert-chain CDDL; url: section-3.3
+        text: Signature header; url: section-3.1
+spec: draft-ietf-httpbis-header-structure; urlPrefix: https://tools.ietf.org/html/draft-ietf-httpbis-header-structure-07#
+    text: Parsing HTTP1 Header Fields into Structured Headers; type: dfn; url: section-4.2
+spec: SHA2; urlPrefix: http://csrc.nist.gov/publications/fips/fips180-4/fips-180-4.pdf
+    type: dfn
+        text: SHA-256; url: #
+</pre>
+<pre class='link-defaults'>
+spec:fetch; type:dfn; for:/; text:response
+spec:streams; type:interface; text:ReadableStream
+</pre>
+
+# Introduction # {#intro}
+
+<em>This section is non-normative.</em>
+
+The Signed Exchanges specification
+[[draft-yasskin-http-origin-signed-responses]] describes a way to provide one or
+more signatures for an HTTP exchange and to check whether any of those
+signatures is trusted as authoritative for a particular origin. This
+specification describes how web browsers load those exchanges. It is expressed
+as several monkeypatches to the [[FETCH]] specification which call algorithms
+defined here.
+
+## Overview ## {#overview}
+
+When fetching a resource (`https://distributor.example.org/foo.sxg`) with the
+`application/signed-exchange` [=MIME type=], the UA parses it, checks its
+signatures, and then if all is well, redirects to its request URL
+(`https://publisher.example.org/foo`) with a "stashed" exchange attached to the
+request. The redirect applies all the usual processing, and then when it would
+normally check for an HTTP cache hit, it also checks whether the stashed request
+matches the redirected request and which of the stashed exchange or HTTP cache
+contents is newer. If the stashed exchange matches and is newer, the UA returns
```

Comment:
This means that whenever UA loads things from a Signed Exchange UA also needs to check HTTP cache. Can you clarify what situation we are trying to deal with by this? If we don't cache the stashed exchange it feels it's just fine, or a site should probably set a short expiration date?

Reply:
Imagine that Vashti visits https://app.com/ regularly, and app.com also publishes a signed exchange valid for a week to allow offline distribution. Now app.com discovers a vulnerability and patches it, but the vulnerable signed exchange is still out there.

If Vashti visits the site after the update and gets the new version into her cache, even stale, then even if an attacker manages to serve her the vulnerable signed exchange, she won't use it.

This isn't an absolute defense against anything, but it lowers the probability of a successful attack on frequent users of a site.

Reply:
Chrome's current impl is not following this, which is the biggest concern, but that part probably needs a separate discussion anyways to proceed. (I'll file an impl bug and take a closer look if this needs to block our further milestones)